docs/knn/index.md

```diff
@@ -4,290 +4,14 @@ title: KNN
 nav_order: 50
 has_children: true
 has_toc: false
-has_math: true
 ---
 
 # k-NN
 
-Short for *k-nearest neighbors*, the k-NN plugin enables users to search for the k-nearest neighbors to a query point across an index of points. To determine the neighbors, a user can specify the space (the distance function) they want to use to measure the distance between points. Currently, the k-NN plugin supports Euclidean, cosine similarity, and Hamming bit spaces.
+Short for *k-nearest neighbors*, the k-NN plugin enables users to search for the k-nearest neighbors to a query point across an index of vectors. To determine the neighbors, a user can specify the space (the distance function) they want to use to measure the distance between points.
 
-Use cases include recommendations (for example, an "other songs you might like" feature in a music application), image recognition, and fraud detection. For background information on the algorithm, see [Wikipedia](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm).
+Use cases include recommendations (for example, an "other songs you might like" feature in a music application), image recognition, and fraud detection. For background information on the k-NN search, see [Wikipedia](https://en.wikipedia.org/wiki/Nearest_neighbor_search).
 
-The k-NN plugin supports two methods of k-NN search. The first method uses the HNSW algorithm to return the approximate k-nearest neighbors. This algorithm sacrifices indexing speed and search accuracy in return for lower latency and more scalable search. To learn more about the algorithm, please refer to [nmslib's documentation](https://github.com/nmslib/nmslib/) or [the paper introducing the algorithm](https://arxiv.org/abs/1603.09320). Currently, only the Euclidean and cosine similarity spaces are available for this method.
+This plugin supports three different methods for obtaining the k-nearest neighbors from an index of vectors. The first method takes an approximate nearest neighbor approach; it uses the HNSW algorithm to return the approximate k-nearest neighbors to a query vector. This algorithm sacrifices indexing speed and search accuracy in return for lower latency and more scalable search. To learn more about the algorithm, please refer to [nmslib's documentation](https://github.com/nmslib/nmslib/) or [the paper introducing the algorithm](https://arxiv.org/abs/1603.09320). The second method extends Elasticsearch's script scoring functionality to execute a brute force, exact k-NN search. With this approach, users are able to run k-NN search on a subset of vectors in their index (sometimes referred to as a pre-filter search). The third method adds the distance functions as painless extensions that can be used in more complex combinations. 
```

Comment:
how about we classify approaches to 2 sections.

1. **Approximate k-NN**
        Description: blah.. blah..
        usecases: <couple sentences on when to apply this> 
        For more details refer https://github.com/jmazanec15/for-elasticsearch-docs/blob/knn-docs-refactor/docs/knn/approximate-knn.md
       
2. **Exact k-NN/Dynamic k-NN/custom scoring**
        Description: blah.. blah..
        usecases: <couple sentences on when to apply this> 
        For more details refer https://github.com/jmazanec15/for-elasticsearch-docs/blob/knn-docs-refactor/docs/knn/knn-score-script.md

3. **Exact k-NN/Dynamic k-NN/custom scoring with painless** 
    Description: This is extension to second method. In this approach the distance functions as painless extensions that can be used in more complex combinations. 
   usecases: <couple sentences on when to apply this> 
   For more details refer https://github.com/jmazanec15/for-elasticsearch-docs/blob/knn-docs-refactor/docs/knn/painless-functions.md

Reply:
Makes sense. I will update.